catalog/pom.xml

```diff
@@ -17,7 +17,6 @@
   <modules>
     <module>citrus-core</module>
     <module>citrus-endpoint-catalog</module>
-    <module>citrus-bom</module>
```

Comment:
we can still keep this as a submodule also when there is no parent relationship anymore. this way we make sure the BOM is built with the project and that the BOM gets published to Maven central during the release process.